backend/clap/src/main/java/zpi/ppea/clap/security/FirebaseAgent.java

```diff
@@ -12,64 +12,56 @@
 import lombok.Data;
 import lombok.RequiredArgsConstructor;
 import lombok.SneakyThrows;
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Service;
-import zpi.ppea.clap.exceptions.NoAccessToResource;
+import zpi.ppea.clap.config.ValueConfig;
+import zpi.ppea.clap.exceptions.GrpcDatastoreException;
 import zpi.ppea.clap.exceptions.UserNotAuthorizedException;
 import zpi.ppea.clap.service.DataStoreService;
 
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Optional;
 
+@Slf4j
 @Service
 @RequiredArgsConstructor
 public class FirebaseAgent {
 
-    private final Logger log = LogManager.getLogger("firebase-app");
     private final DataStoreService client;
+    private final ValueConfig valueConfig;
+    public static final String REFRESH_TOKEN_NAME = "refresh";
 
     @SneakyThrows
     @PostConstruct
     public void init() {
         FirebaseOptions options = null;
-        try (var input = this.getClass()
-                .getClassLoader()
-                .getResourceAsStream("service-account.json")) {
-
+        try (var input = this.getClass().getClassLoader().getResourceAsStream("service-account.json")) {
             if (input != null)
-                options = FirebaseOptions.builder()
-                    .setCredentials(GoogleCredentials.fromStream(input))
-                    .build();
+                options = FirebaseOptions.builder().setCredentials(GoogleCredentials.fromStream(input)).build();
         }
 
-        if (options == null) {
-            throw new NoAccessToResource(new Exception("failed to initialize firebase app"), "");
-        }
+        if (options == null)
+            throw new GrpcDatastoreException(new Exception("Failed to initialize firebase app"), "");
```

Comment:
That's not really a Grpc related issue. Perviously since, we could not initialze app most likely `"service-account.json"` was missing, hence `NoAccessToResource`

Reply:
NoAccessToResource was thrown as well if the grpc server was down or grpc method was not implemented. Should we throw NoAccessToResource in these Situatuions as well? Shouldn't we add a new error here?

Reply:
No access to resource is broad enough to cover all those topics. If database failed - we couldn't access resource, if the transport over grpc failed we couldn't access resource etc. 
I don't think this exception is appropriate here, so yes, we could use new type of excpetion